### PR TITLE
[docs] Add a deployment FAQ section to Orc8r docs

### PR DESCRIPTION
**Pull request by karthiksubraveti**
Signed-off-by: Karthik Subraveti <ksubraveti@fb.com>

## Summary

- Add deployment types section and add FAQ section to orc8r deployment
Added some failures documented by @hcgatewood in (Orc8r Ux improvements) in this doc. 

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: docs/readmes/orc8r/deploy_faq.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+kubectl -n orc8r describe pods
+# pod status ImagePullBackOff, indicates that Image wasn't found
+```
+
+* Get the helm’s view of images
+
+```
+`helm version
+
+helm list
```
Comment (mpgermano) [inline docs/readmes/orc8r/deploy_faq.md]: I think you need a `-n orc8r` here. So it's `helm -n orc8r list`

File: docs/readmes/orc8r/deploy_faq.md [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+* Check if there is an issue with helm repo url, password or container image tag
+* Ensure container image registry and Terraform values are correct
+
+* Run following kubectl commands to get more details on the error encountered
+
+```
+kubectl -n orc8r get pods
+kubectl -n orc8r describe pods
+# pod status ImagePullBackOff, indicates that Image wasn't found
```
Comment (mpgermano) [inline docs/readmes/orc8r/deploy_faq.md:30]: I think it's also worth noting that a pod could be crash looping. I've often seen this cause timeouts.

File: docs/readmes/orc8r/deploy_faq.md [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+        template was:
+                {.items[0].metadata.name}
+        object given to jsonpath engine was:
+                map[string]interface {}{"apiVersion":"v1", "items":[]interface {}{}, "kind":"List", "metadata":map[string]interface {}{"resourceVersion":"", "selfLink":""}}
+error: pod, type/name or --filename must be specified
+```
+
+**Resolution Steps**
+
+* Problem was that deploy_nms wasn’t set to true in variables.tf. Set deploy_nms to true and rerun `terraform apply`
```
Comment (mpgermano) [inline docs/readmes/orc8r/deploy_faq.md]: Nit: `The problem`

File: docs/readmes/orc8r/deploy_faq.md [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+## Error: ValidationException: Domain is being deleted for aws_elasticsearch_domain_policy
+
+```
+`Error: ValidationException: Domain is being deleted` for `aws_elasticsearch_domain_policy`
+
+```
+
+**Resolution Steps**
+
+* Sometimes ES domains aren’t deleted immediately, so need to change ES domain to new value and stop tracking old domain
```
Comment (mpgermano) [inline docs/readmes/orc8r/deploy_faq.md]: `aren't deleted immediately. Change the ES domain to a new value...`

**Review by mpgermano — APPROVED** [review]
Looks good. A few comments

File: docs/readmes/orc8r/deploy_faq.md [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+```
+helm version (provides current version of helm)
+helm -n orc8r list (provides the list of releases which are currently deployed under orc8r namespace)
+helm -n orc8r get values orc8r (gets the values file for the orc8r release)
+```
+
+## NMS pod wasn’t showing up properly
+
+```
+kubectl exec -it $(kubectl get pod -l app.kubernetes.io/component=magmalte -o jsonpath='{.items[0].metadata.name}') -- yarn setAdminPassword master xxxx@xxxx.com 1234
```
Comment (sudhikan) [inline docs/readmes/orc8r/deploy_faq.md:46]: missing the `-n orc8r` in the kubectl command.

**Pull request merged**